### DISCUSSION TITLE
async-pre-read not working with combi

### DISCUSSION THREAD

**Discussion by alperyilmaz** [davatorium/rofi#1819] [(Possible) Issues]
Hi,
Let's generate a `slow.sh` script:
```bash
#!/usr/bin/bash

for i in $(seq 10); do
  echo $i
  sleep 1.5
done
```

If I run the script and send it thru pipe, rofi works async (even without `-async-pre-read` option), rofi immediately opens up GUI and list is updated as script produces output.
```bash
./slow.sh | rofi -dmenu
```

I want to use the async property in combi mode, even with `-async-pre-read` option, rofi waits for slow script to finish in order to show the UI.
```bash
rofi -modi combi -combi-modi drun,"slow:./slow.sh" -show combi -config /dev/null -async-pre-read 0
```

is there a way to have async functionality in combi mode? my rofi version is 1.7.5-81-git-d786129f

**Comment by DaveDavenport** [comment]
- Your version does not have the `-async-pre-read` option.
- - When the option existed in the past it *only* existed for dmenu emulation.
- combi mode combines other modes, it has no functionality itself. An 'async-pre-read' option does not make sense for it. Do you mean script mode?-

**Comment by alperyilmaz** [comment]
Sorry if I misused some terms. In that case, yes I mean "script mode". A script is running slow and its output should not block rofi UI start. I noticed `rofi-blocks` but it's not a native solution I guess. 
What type of approach do you suggest in this case? Can this situation be solved by an advanced bash script?

**Reply by DaveDavenport** [reply]
It is something that needs to be fixed on rofi side.  I have not had time to look at this.
`rofi-blocks` might be a nice solution , script mode is very simple. rofi-blocks offers more options.
